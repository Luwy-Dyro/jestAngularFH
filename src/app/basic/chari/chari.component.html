<h1>Poke</h1>

<div *ngIf="!chari; else elseBlock">
    <h2>Loading...</h2>
</div>

<ng-template #elseBlock>

    <h3>{{chari?.name | titlecase}} </h3>
    <img [src]="chari?.sprites?.front_default" [alt]="chari?.name">

</ng-template>