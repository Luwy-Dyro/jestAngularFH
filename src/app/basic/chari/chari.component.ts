import { Component, OnInit } from '@angular/core';
import { PokeService } from '../services/poke.service';
import { Poke } from '../interfaces/poke';

@Component({
  selector: 'app-chari',
  templateUrl: './chari.component.html',
  styleUrls: ['./chari.component.scss']
})
export class ChariComponent implements OnInit {

  public chari?: Poke 

  constructor(private poService: PokeService ){

  }


  ngOnInit() {
    
      this.poService.getPoke(6)
          .subscribe( resp => {

            this.chari = resp
            console.log(resp);
            

          })

  }

  
}
