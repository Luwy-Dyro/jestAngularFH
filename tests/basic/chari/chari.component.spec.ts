import { ComponentFixture, TestBed } from '@angular/core/testing';
import { HttpClientTestingModule, HttpTestingController }  from '@angular/common/http/testing'
import { ChariComponent } from '../../../src/app/basic/chari/chari.component';
import { PokeService } from '../../../src/app/basic/services/poke.service';


describe('ChariComponent', () => {
  let component: ChariComponent;
  let fixture: ComponentFixture<ChariComponent>;
  let compiled:HTMLElement;
  let servicePok: PokeService;
  let httpMock: HttpTestingController;


  beforeEach(async () => {
    await TestBed.configureTestingModule({
      declarations: [ ChariComponent ],
      imports: [HttpClientTestingModule],
      providers: [PokeService]
    })
    .compileComponents();

    fixture = TestBed.createComponent(ChariComponent);
    component = fixture.componentInstance;
  
    servicePok = TestBed.inject(PokeService);
    httpMock = TestBed.inject(HttpTestingController)

    fixture.detectChanges();
    compiled = fixture.nativeElement
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });

  it('debe hacer match Snapshot', () => {
    //console.log(compiled.innerHTML);  
    expect(compiled.innerHTML).toMatchSnapshot();
  });


  it('debe de mostrar un loading al inicio', () => {

    const h2 = compiled.querySelector('h2')   
    expect(h2?.textContent).toContain('Loading...');
  });


  it('debe de cargar Chari de inmediato', () => {

    const dummyPoke = {
      name: 'Charizard',
      sprites:{
        front_default: 'https://charizard.com/sprite.png'
      }
    }

    const request = httpMock.expectOne('https://pokeapi.co/api/v2/pokemon/6')
    
    expect(request.request.method).toBe('GET');

    request.flush(dummyPoke);

    //ciclo de vida ang, regenera los omponentes,
    fixture.detectChanges()
    //console.log(compiled.innerHTML);
    
    const h3 = compiled.querySelector('h3');
    const img = compiled.querySelector('img');

    expect(h3?.textContent?.toLocaleLowerCase() ).toContain(dummyPoke.name.toLocaleLowerCase() )

    expect( img?.src ).toBe(dummyPoke.sprites.front_default)
    //expect( img?.alt ).toBe(dummyPoke.name + '!!!')
    expect( img?.alt ).toBe(dummyPoke.name)

  });


});
