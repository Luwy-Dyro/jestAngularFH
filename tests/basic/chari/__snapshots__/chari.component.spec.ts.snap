// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChariComponent debe hacer match Snapshot 1`] = `
"<h1>Poke</h1><div><h2>Loading...</h2></div><!--bindings={
  "ng-reflect-ng-if-else": "[object Object]"
}--><!--container-->"
`;
